src/meta/coinductive_predicates.lean

```diff
@@ -84,11 +84,25 @@ meta def mk_and_lst : list expr → expr := mk_op_lst `(and) `(true)
 
 meta def mk_or_lst : list expr → expr := mk_op_lst `(or) `(false)
 
-meta def elim_gen_prod : nat → expr → list expr → tactic (list expr × expr)
-| 0       e hs := return (hs, e)
-| (n + 1) e hs := do
-  [(_, [h, h'], _)] ← induction e [],
-  elim_gen_prod n h' (hs ++ [h])
+meta def mk_sigma : list expr → tactic expr
```

Comment:
I promise my constant requests for documentation aren't to be annoying. I don't know what this function is supposed to do. I tried it on some inputs (``[`(1=1), `(2=2)]``, ``[`(1), `(2=2)]``, etc) and got app-builder exceptions. From the call to `lambdas`, the input is probably expected to be a list of local constants? Is that required? What's the intended failure behavior?

Reply:
```suggestion
meta def mk_psigma : list expr → tactic expr
```
Thanks for clarifying. To improve the failure message, how about changing the patterns in 92 and 93 to `x@(expr.local_const _ _ _)`, and adding a catchall case `| _ := fail "mk_sigma expects a list of local constants"`?